adafruit_circuitplayground/circuit_playground_base.py

```diff
@@ -689,23 +690,38 @@ def red_led(self, value):
     @staticmethod
     def _sine_sample(length):
         tone_volume = (2**15) - 1
+        # Amplitude shift up in order to not have negative numbers
         shift = 2**15
         for i in range(length):
             yield int(tone_volume * math.sin(2 * math.pi * (i / length)) + shift)
 
-    def _generate_sample(self, length=100):
+    @staticmethod
+    def _square_sample(length):
+        # Square waves are MUCH louder than then sine
+        tone_volume = (2**16) - 1
+        half_length = length // 2
+        for _ in range(half_length):
+            yield tone_volume
+        for _ in range(half_length):
+            yield 0
+
+    def _generate_sample(self, length=100, waveform="sine"):
         if self._sample is not None:
             return
-        self._sine_wave = array.array("H", self._sine_sample(length))
+        if waveform == "square":
+            self._wave = array.array("H", self._square_sample(length))
+        else:
+            self._wave = array.array("H", self._sine_sample(length))
```

Comment:
This may shrink things enough to fit
```suggestion
        self._wave = array.array("H", self._square_sample(length) if waveform == "square" else self._sine_sample(length))
```

Reply:
There may be similar refactorings in the Python code that would save a few bytes here and there.

Reply:
I wasn't sure if ternary  operator worked in Circuit Python.  I wrote it this way if anyone else wanted to add different shaped waves down the line.  If it works, great but to me the sound difference isn't that big.  Square definitely sounds more ragged than sine but that could be that the code is driving it rather hard.